### PR TITLE
ESPCN on ADNI Dataset

### PR DESCRIPTION
**Pull request by zharbutt**
Implementation of the ESPCN on ADNI dataset

Files:
modules.py
dataset.py
train.py
predict.py
README.md

### CODE REVIEW

**Comment by uqmarlonbran** [comment]
**This is an initial inspection - no action is required yet**

**Difficulty:** Normal

**General Comments:** Good approach overall. Custom dataloader and metrics computed over the test set (and a separate validation set). Only saves the model with the lowest validation loss which is good. README a bit light on details (e.g. file structure), but otherwise ok.

**[Recognition Problem]**
"modules.py":

- Re-implementation of the Keras example in PyTorch. All looks ok.

"dataset.py":

- Nice implementation of custom dataset to deal with AD and NC data structure. 
- Creates a training and validation set.

"train.py":

- Correctly keeps track of training and validation loss.
- Saves the model with the lowest validation loss. Nice.
- Creates training/validation loss plots.
- Computes metrics on the test set using PSNR.
- Provides a visualization of the reconstructed images.

"predict.py":

- Correctly loads the trained weights.
- Provides a visualization, but no metrics (though metrics are computed in the train script).

"README.MD":

- Brief description of the network.
- Reconstructions look ok, in line with other attempts.
- Training plot looks ok.
- Dependencies provided.

**[Commit Log]**
Commit Log looks ok.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, could visualise metrics -1
Module present
Commenting minimal -1
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable but missing info like usage, data etc. -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description minimal -1
